String shrinking is not being executed

See ShrinkingTest.kt for an example.

I have a similiar issue, here a short reproducer:


My code

```kotlin
import io.kotest.core.spec.style.StringSpec
import io.kotest.property.PropTestConfig
import io.kotest.property.PropertyTesting
import io.kotest.property.ShrinkingMode
import io.kotest.property.forAll

class PropertyExample: StringSpec({
    "Shrinking happens" {
        PropertyTesting.shouldPrintShrinkSteps = true
        val config = PropTestConfig(shrinkingMode = ShrinkingMode.Unbounded)
        forAll<Int>(config) { a ->
            a < 1000
        }
    }
}
```

Output:
```
Property failed after 4 attempts

	Arg 0: 2147483647

Repeat this test by using seed 776132071643096315

Caused by AssertionFailedError: expected:<true> but was:<false>
Expected :true
Actual   :false
<Click to see difference>

java.lang.AssertionError: Property failed after 4 attempts

	Arg 0: 2147483647

Repeat this test by using seed 776132071643096315
```

So as you can see, no shrinking happened, even though I have set the ShrinkingMode to Unbounded.

In the Kotest code I can see that an IntShrinker exists and is used for the Integers Arb, but somehow it is not executed:
```kotlin
fun Arb.Companion.int(range: IntRange = Int.MIN_VALUE..Int.MAX_VALUE): Arb<Int> {
   val edgecases = listOf(0, 1, -1, Int.MAX_VALUE, Int.MIN_VALUE).filter { it in range }
   return arbitrary(edgecases, IntShrinker(range)) { it.random.nextInt(range) }
}
```

--- a/kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/builders.kt
+++ b/kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/builders.kt
@@ -27,7 +27,7 @@ fun <A> arbitrary(edgecases: List<A>, fn: (RandomSource) -> A): Arb<A> = object 
 fun <A> arbitrary(edgecases: List<A>, shrinker: Shrinker<A>, fn: (RandomSource) -> A): Arb<A> = object : Arb<A>() {
    override fun edgecases(): List<A> = edgecases
    override fun sample(rs: RandomSource): Sample<A> = sampleOf(fn(rs), shrinker)
-   override fun values(rs: RandomSource): Sequence<Sample<A>> = generateSequence { Sample(fn(rs)) }
+   override fun values(rs: RandomSource): Sequence<Sample<A>> = generateSequence { sampleOf(fn(rs), shrinker) }
 }
 
 /**

--- a/kotest-property/src/jvmTest/kotlin/com/sksamuel/kotest/property/shrinking/ShrinkingTest.kt
+++ b/kotest-property/src/jvmTest/kotlin/com/sksamuel/kotest/property/shrinking/ShrinkingTest.kt
@@ -10,7 +10,7 @@ import io.kotest.property.checkAll
 
 class ShrinkingTest : FunSpec() {
    init {
-      test("!shrinking should show the exception raised by the shrunk values") {
+      test("shrinking should show the exception raised by the shrunk values") {
 
          val stdout = captureStandardOut {
             shouldThrowAny {

